>From 309ac5b2ef96678f30c67b71e342813688040e72 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 4 Jul 2012 17:19:20 +0200
Subject: [PATCH 01/15] buildsys: Move msi[x] and virtio-pci from
 Makefile.objs to Makefile.target

Building both kvm and non kvm versions of qemu in the same tree,
results in a link error during the non kvm build with undefined references
to kvm_kernel_irqchip, expanded from the kvm_irqchip_in_kernel macro.

This patch fixes this.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 Makefile.objs   |    2 --
 Makefile.target |    3 ++-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile.objs b/Makefile.objs
index 264f1fe..8d49738 100644
--- a/Makefile.objs
+++ b/Makefile.objs
@@ -211,10 +211,8 @@ hw-obj-y =
 hw-obj-y += vl.o loader.o
 hw-obj-$(CONFIG_VIRTIO) += virtio-console.o
 hw-obj-y += usb/libhw.o
-hw-obj-$(CONFIG_VIRTIO_PCI) += virtio-pci.o
 hw-obj-y += fw_cfg.o
 hw-obj-$(CONFIG_PCI) += pci_bridge.o pci_bridge_dev.o
-hw-obj-$(CONFIG_PCI) += msix.o msi.o
 hw-obj-$(CONFIG_PCI) += shpc.o
 hw-obj-$(CONFIG_PCI) += slotid_cap.o
 hw-obj-$(CONFIG_PCI) += pci_host.o pcie_host.o
diff --git a/Makefile.target b/Makefile.target
index eda8637..ede8ed3 100644
--- a/Makefile.target
+++ b/Makefile.target
@@ -183,9 +183,10 @@ obj-y = arch_init.o cpus.o monitor.o machine.o gdbstub.o balloon.o ioport.o
 # virtio has to be here due to weird dependency between PCI and virtio-net.
 # need to fix this properly
 obj-$(CONFIG_NO_PCI) += pci-stub.o
-obj-$(CONFIG_PCI) += pci.o
+obj-$(CONFIG_PCI) += pci.o msi.o msix.o
 obj-$(CONFIG_VIRTIO) += virtio.o virtio-blk.o virtio-balloon.o virtio-net.o virtio-serial-bus.o
 obj-$(CONFIG_VIRTIO) += virtio-scsi.o
+obj-$(CONFIG_VIRTIO_PCI) += virtio-pci.o
 obj-y += vhost_net.o
 obj-$(CONFIG_VHOST_NET) += vhost.o
 obj-$(CONFIG_REALLY_VIRTFS) += 9pfs/virtio-9p-device.o
-- 
1.7.10.4


